sycl/source/detail/pi.cpp

```diff
@@ -452,7 +452,8 @@ static void initializePlugins(std::vector<plugin> &Plugins) {
       std::cerr << "SYCL_PI_TRACE[basic]: "
                 << "Plugin found and successfully loaded: "
                 << PluginNames[I].first << " [ PluginVersion: "
-                << Plugins[I].getPiPlugin().PluginVersion << "]" << std::endl;
+                << Plugins[PluginIdx++].getPiPlugin().PluginVersion << " ]"
```

Comment:
Could we use `Plugins.end()` instead of `PluginIdx`? Alternatively, you could change the lines further above to something like

```c++
plugin &NewPlugin = Plugins.emplace_back(PluginInformation, PluginNames[I].second, Library);
```

and then use `NewPlugin` instead.

Reply:
Thanks. Change made